ravedude/src/avrdude/mod.rs

```diff
@@ -17,6 +17,50 @@ pub struct Avrdude {
 }
 
 impl Avrdude {
+    pub fn require_min_ver((req_major, req_minor): (u8, u8)) -> anyhow::Result<()> {
```

Comment:
I think you can make this function more manageable by splitting it into two parts:  The parsing part and the reporting part.  Notice how I dropped all the `.context()` calls in favor of a single one when invoking the `get_avrdude_version()` function.  A user won't care about the exact failure, we'll just need to tell them that _something_ didn't work.  They can piece together the rest from the error trace.

```rust
    fn get_avrdude_version() -> anyhow::Result<(u8, u8)> {
        let output = process::Command::new("avrdude").arg("-?").output()?;
        let stderr: &str = std::str::from_utf8(&output.stderr)?;
        let version: &str = stderr.split("version").last()?.trim();
        let (version, _) = version.split_once('-')?.to_owned();
        let (major, minor) = version.split_once('.').unwrap();
        let major = major.parse::<u8>()?;
        let minor = minor.parse::<u8>()?;
        (major, minor)
    }

    pub fn require_min_ver((req_major, req_minor): (u8, u8)) -> anyhow::Result<()> {
        let (major, minor) = Self::get_avrdude_version()
            .context("Failed reading avrdude version information")?;
        if major < req_major {
            // ...
        }
        Ok(())
    }
```

Reply:
Yes, much easier on the eye. And you are right, while the context messages were helpful during implementing and testing, in the end, the user won't care that much.

Reply:
In `get_avrdude_version()` I had to introduce `ok_or_else(err)` because `last()` yields an `Option` while we need a `Result`.